src/Blumchen/Subscriber/OptionsBuilder.cs

```diff
@@ -0,0 +1,188 @@
+
+
+        _replicationDataMapperSelector.Add(WildCard,
+            new Tuple<IReplicationJsonBMapper, IMessageHandler, MethodInfo>(ObjectReplicationDataMapper.Instance, handler, methodInfo));
+        return this;
+    }
+
+    private OptionsBuilder ConsumesRaw<T, TU>(IMessageHandler<T> handler,
+        IReplicationJsonBMapper dataMapper) where T : class where TU : class
+    {
+        var urns = typeof(T)
+            .GetCustomAttributes(typeof(RawUrnAttribute), false)
+            .OfType<RawUrnAttribute>()
+            .Select(attribute => attribute.Urn).ToList();
+        Ensure.RawUrn<IEnumerable<Uri>,T>(urns, nameof(NamingPolicy));
+
+        var methodInfo = handler
+                             .GetType()
+                             .GetMethod(nameof(IMessageHandler<TU>.Handle), BindingFlags.Instance | BindingFlags.Public, [typeof(TU)])
+                         ?? throw new ConfigurationException($"Unable to find {nameof(IMessageHandler<TU>)} implementation on {handler.GetType().Name}");
+
+        using var urnEnum = urns.GetEnumerator();
+        while (urnEnum.MoveNext())
+            _replicationDataMapperSelector.Add(urnEnum.Current.ToString(),
+                new Tuple<IReplicationJsonBMapper, IMessageHandler, MethodInfo>(dataMapper, handler, methodInfo));
+        return this;
+    }
+
+    [UsedImplicitly]
+    public OptionsBuilder WithErrorProcessor(IErrorProcessor? errorProcessor)
+    {
+        _errorProcessor = errorProcessor;
+        return this;
+    }
+
+    internal ISubscriberOptions Build()
+    {
+        _tableDescriptor ??= TableDescriptorBuilder.Build();
+        Ensure.NotNull<NpgsqlConnectionStringBuilder?>(_connectionStringBuilder, $"{nameof(ConnectionString)}");
+        Ensure.NotNull<NpgsqlDataSource?>(_dataSource, $"{nameof(DataSource)}");
+
+        if (_typeRegistry.Count > 0)
+        {
+            Ensure.NotNull(_namingPolicy, $"{nameof(NamingPolicy)}");
+            if (_jsonSerializerContext != null)
+            {
+                var typeResolver = new JsonTypeResolver(_jsonSerializerContext, _namingPolicy);
+                foreach (var type in _typeRegistry.Keys)
+                    typeResolver.WhiteList(type);
+
+                _jsonDataMapper =
+                    new JsonReplicationDataMapper(typeResolver, new JsonReplicationDataReader(typeResolver));
+
+                foreach (var (key, value) in typeResolver.RegisteredTypes.Join(_typeRegistry, pair => pair.Value,
+                             pair => pair.Key, (pair, valuePair) => (pair.Key, valuePair.Value)))
+                    _replicationDataMapperSelector.Add(key,
+                        new Tuple<IReplicationJsonBMapper, IMessageHandler, MethodInfo>(_jsonDataMapper, value.Item1, value.Item2));
+            }
+            else
+            {
+                throw new ConfigurationException($"`${nameof(Consumes)}<>` requires a valid `{nameof(JsonSerializerContext)}`.");
```

Comment:
**CONCERN:** Could you expand on this part? I understand that your preference is to use source generators for JSON serialisation. I agree that's the most performant and recommended way, but for many people having default json serialisation will be more than enough.

Reply:
I have a check the AOT part. Nevertheless I can abstract further that part to provide additional options